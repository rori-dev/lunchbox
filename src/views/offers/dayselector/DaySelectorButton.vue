<template>
  <button
    class="text-primary-500 hover:text-primary-600
           disabled:text-primary-300 disabled:cursor-not-allowed"
    :class="{ 'active:text-primary-800': !disabled }"
    :title="title"
    :aria-label="title"
    @click="clicked"
    :disabled="disabled"
    :aria-keyshortcuts="isPrevious ? 'ArrowLeft' : 'ArrowRight'"
  >
    <AngleLeftIcon v-if="isPrevious" class="w-16 h-16" />
    <AngleRightIcon v-else class="w-16 h-16" />
  </button>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import AngleLeftIcon from '@/assets/icons/angle-left.svg'
import AngleRightIcon from '@/assets/icons/angle-right.svg'
import { DaySelectorDirection } from '@/views/offers/dayselector/DaySelectorEvent'

@Component({
  components: {
    AngleLeftIcon,
    AngleRightIcon,
  },
})
export default class DaySelectorButton extends Vue {
  @Prop() disabled!: boolean
  @Prop() direction!: DaySelectorDirection

  clicked() {
    this.$emit('click')
  }

  get isPrevious(): boolean {
    return this.direction === DaySelectorDirection.PREVIOUS
  }

  get title(): string {
    return this.isPrevious ? 'Zu vorherigem Tag wechseln' : 'Zu n√§chstem Tag wechseln'
  }
}
</script>
