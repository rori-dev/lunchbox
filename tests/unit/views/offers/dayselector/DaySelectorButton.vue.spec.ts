import DaySelectorButton from '@/views/offers/dayselector/DaySelectorButton.vue'
import { mountUnit } from '@tests/unit/test-util'

describe('DaySelectorButton', () => {
  it('renders snapshot for previous day', () => {
    const wrapper = mountUnit(DaySelectorButton, {
      direction: 'prev',
      disabled: false,
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders snapshot for next day', () => {
    const wrapper = mountUnit(DaySelectorButton, {
      direction: 'next',
      disabled: false,
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('has no active state when disabled', () => {
    const wrapper = mountUnit(DaySelectorButton, {
      direction: 'prev',
      disabled: true,
    })

    expect(wrapper.classes().find(c => c.startsWith('active'))).toBeFalsy()
  })
})
