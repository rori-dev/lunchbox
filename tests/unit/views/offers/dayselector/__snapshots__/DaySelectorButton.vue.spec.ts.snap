// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaySelectorButton renders snapshot for next day 1`] = `
<button
  aria-keyshortcuts="ArrowRight"
  aria-label="Zu nächstem Tag wechseln"
  class="text-primary-500 hover:text-primary-600
         disabled:text-primary-400 disabled:opacity-25 disabled:cursor-not-allowed active:text-primary-800"
  title="Zu nächstem Tag wechseln"
>
  <anglerighticon-stub
    class="w-16 h-16"
  />
</button>
`;

exports[`DaySelectorButton renders snapshot for previous day 1`] = `
<button
  aria-keyshortcuts="ArrowLeft"
  aria-label="Zu vorherigem Tag wechseln"
  class="text-primary-500 hover:text-primary-600
         disabled:text-primary-400 disabled:opacity-25 disabled:cursor-not-allowed active:text-primary-800"
  title="Zu vorherigem Tag wechseln"
>
  <anglelefticon-stub
    class="w-16 h-16"
  />
</button>
`;
